biz.aQute.repository/test/aQute/bnd/repository/maven/pom/provider/PomRepositoryTest.java

```diff
@@ -94,14 +94,51 @@ public void testBndPomRepoFile() throws Exception {
 		assertEquals(1, list.size());
 	}
 
+	public void testBndPomRepoFileExistingParent() throws Exception {
+		BndPomRepository bpr = new BndPomRepository();
+		Workspace w = Workspace.createStandaloneWorkspace(new Processor(), tmp.toURI());
+		w.setBase(tmp);
+		bpr.setRegistry(w);
+
+		Map<String,String> config = new HashMap<>();
+		config.put("pom", "testdata/pomrepo/existing-parent.xml");
+		config.put("snapshotUrls", "https://repo1.maven.org/maven2/");
+		config.put("releaseUrls", "https://repo1.maven.org/maven2/");
+		config.put("name", "test");
+		bpr.setProperties(config);
+
+		List<String> list = bpr.list(null);
+		assertNotNull(list);
+		assertEquals(1, list.size());
+	}
+
+	public void testBndPomRepoFileMissingParent() throws Exception {
+		BndPomRepository bpr = new BndPomRepository();
+		Workspace w = Workspace.createStandaloneWorkspace(new Processor(), tmp.toURI());
+		w.setBase(tmp);
+		bpr.setRegistry(w);
+
+		Map<String,String> config = new HashMap<>();
+		config.put("pom", "testdata/pomrepo/missing-parent.xml");
+		config.put("snapshotUrls", "https://repo1.maven.org/maven2/");
+		config.put("releaseUrls", "https://repo1.maven.org/maven2/");
+		config.put("name", "test");
+		bpr.setProperties(config);
+
+		try {
+			bpr.list(null);
+			fail("Should throw IllegalArgumentException on missing parent pom.");
+		} catch (IllegalArgumentException iae) {}
```

Comment:
Perhaps IllegalArgumentException is too specific? Maybe just `catch (Exception)` would be better. I would also put a comment in the catch block to indicate an exception is expected behavior.